src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1118,23 +1118,36 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             Validity.Assert(svThisPtr.IsPointer,
                             "this pointer wasn't a pointer. {89635B06-AD53-4170-ADA5-065EB2AE5858}");
 
-            int typeID = svThisPtr.metaData.type;
-
-            //Test for exact match
-            List<FunctionEndPoint> exactFeps = new List<FunctionEndPoint>();
-
-            foreach (FunctionEndPoint fep in feps)
-                if (fep.ClassOwnerIndex == typeID)
-                    exactFeps.Add(fep);
-
-            if (exactFeps.Count == 1)
-            {
-                return exactFeps[0];
-            }
-
 
+            // We have multiple possible scopes for the function call:
+            // 1. Static method call - no this pointer
+            // ex: ClassA.Method();
+            //    Hidden static methods generate multiple feps.
+            //    We do not need to check actually if the method has the "IsHideBySig" (https://docs.microsoft.com/en-us/dotnet/api/system.reflection.methodbase.ishidebysig)
+            //    because static methods can only be hidden.
+            // 2. Method call from an instance of a class - valid this pointer.
+            // ex: classAInstance.method();
+            // 3. Function from the global scope - no this pointer and no class scope.
+            //    In this case the svThisPtr.metaData.type will hold the global 
+            // ex: SomeGlobalFunction();
+            //
```

Comment:
@aparajit-pratap do you agree with these 3 scenarios ? Did I get them right ?

Reply:
Hmm...from what I've seen in calls stacks...I would agree with you...I will remove that part